examples/scripts/TreeCompareAuto.C

```diff
@@ -0,0 +1,293 @@
+            continue;
+        }
+
+        // if also the histograms are not identical check if the histograms
+        // are equal on a statistical base. Use The Kolmogorov test for
+        // this.
+        outstream << "Checking Kolmogorov" << endl;
+
+        double kolmo = htemp->KolmogorovTest(hist1);
+
+        outstream << "Result of Kolmogorov test: " << kolmo << endl;
+        if (kolmo > 0.99) {
+            numKolmogorovHistograms++;
+            outstream << "**********************" << std::endl;
+            hist1->Clear();
+            hist->Clear();
+            htemp->Clear();
+            continue;
+        }
+
+        outstream << "Data are differnt" << std::endl;
+        outstream << "**********************" << std::endl;
+        outstream << "Entries: " << hist->GetEntries() << std::endl;
+        outstream << "Mean: " << hist->GetMean() << std::endl;
+        outstream << "RMS: " << hist->GetRMS() << std::endl;
+        outstream << "Underflow: " << hist->GetBinContent(0) << std::endl;
+        outstream << "Overflow: " << hist->GetBinContent(hist->GetNbinsX() + 1) << std::endl;
+        outstream << "**********************" << std::endl;
+        okay = false;
+        numFailedLeaves++;
+        hist1->Clear();
+        hist->Clear();
+        htemp->Clear();
+    }
+
+    if (!okay) {
+        std::cout << outstream.str();
+        std::cout << "Test failed." << std::endl;
+        std::cout << numFailedLeaves << " of " << numTestedLeaves << " leaves are different." << std::endl;
+        return 1;
+    }
+
+    //  std::cout << outstream.str();
+    std::cout << "Tested leaves:                    " << numTestedLeaves << std::endl;
+    //  std::cout << "Empty leaves:                     " << numEmptyLeaves << std::endl;
+    std::cout << "Leaves with identical entries:    " << numIdenticalEntries << std::endl;
+    std::cout << "Leaves with identical histograms: " << numIdenticalHistograms << std::endl;
+    std::cout << "Leaves with kolmo histograms:     " << numKolmogorovHistograms << std::endl;
+    std::cout << "Test passed. All leaves of all branches are exactly identical." << std::endl;
+
+    return 0;
+}
+
+bool operator==(TH1 const& lhs, TH1 const& rhs)
+{
+    for (int x = 0; x < lhs.GetNbinsX() + 1; ++x) {
+        if (lhs.GetBinContent(x) != rhs.GetBinContent(x)) {
+            return false;
+        }
+    }
+    return true;
+}
+
+std::vector<TString> GetLeafNames(TTree& cbmsim)
+{
+
+    std::vector<TString> ListOfLeaves;
+
+    TObjArray* bla1 = cbmsim.GetListOfLeaves();
+    TIter myiter1(bla1);
+    TBranch* branch;
+    while ((branch = (TBranch*)myiter1.Next())) {
+        TString mystring = branch->GetName();
+        //      cout << "Branch Name: " << mystring << endl;
+
+        // Generate leaf names for points, digis and hits
+        if (mystring.Contains("Point") || mystring.Contains("Digi") || mystring.Contains("Hit")) {
+            TObjArray* bla2 = mystring.Tokenize(".");
```

Comment:
Regarding the chosen variable names:
* `bla1`
* `myiter1`
* `mystring`
* `bla2`

Please see [S.2](https://github.com/FairRootGroup/FairRoot/blob/dev/CONTRIBUTING.md#s2-prefer-variable-names-describing-what-they-contain)

Reply:
@fuhlig1 
`bla1` -> `list`/`listOfLeaves`
`myiter1` -> `iter`/`listIter`/`leavesIter`
`mystring` -> `branchName`
`bla2` -> `branchNameTokens`

Reply:
Fixed.

Reply:
See also my [rewrite study for inspiration](https://github.com/FairRootGroup/FairRoot/pull/1419#discussion_r1253744499). Imho, we should avoid all those variable and just use this syntax:

```cpp
for (auto leaf : TRangeDynCast<TLeaf>(simTree.GetListOfLeaves())) {
    if (!leaf) {
        continue;
    }
    
    // ...
}
```

* This `nullptr` check is still ugly and error-prone (people will forget it), but that is a consequence of living in the reference semantics world which is a fundamental choice of FairRoot/ROOT...
* The second ugly aspect that we cannot use `auto const leaf` because `GetName()` has no `const`-correct overload.

But besides those, I think this is much safer and more readable than having 3 extra named boilerplate variables for just facilitating a read loop.